vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/WebAtomicActionFactories.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.web.action;
+
+import static org.apache.commons.lang3.Validate.notEmpty;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.apache.commons.lang3.reflect.TypeUtils;
+import org.openqa.selenium.Point;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.vividus.selenium.KeysManager;
+import org.vividus.ui.action.AtomicActionFactory;
+
+public final class WebAtomicActionFactories
+{
+    private WebAtomicActionFactories()
+    {
+    }
+
+    private static void buildKeysActions(KeysManager keysManager, List<String> keys,
+            Consumer<CharSequence> actionBuilder)
+    {
+        notEmpty(keys, "At least one key should be provided");
+        for (String key : keys)
+        {
+            actionBuilder.accept(key.length() == 1 ? key : keysManager.convertToKey(true, key));
```

Comment:
why don't we perform key validation in case of one key?

Reply:
this logic was here before: https://github.com/vividus-framework/vividus/blob/c01f672b1e6fdf12df6b7c335dfb9a769c2fa36f/vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/WebSequenceActionType.java#L198-L213